### PR TITLE
Sync Data mapper code action with latest ballerina api

### PR DESCRIPTION
**Pull request by isuruRuhu**
## Purpose
Sync Data mapper code action with latest ballerina api

## Remarks
TODO 
- Add unit tests. 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java [hunk 1]
```diff
@@ -17,20 +17,7 @@
...
-import io.ballerina.compiler.syntax.tree.MethodCallExpressionNode;
-import io.ballerina.compiler.syntax.tree.NameReferenceNode;
-import io.ballerina.compiler.syntax.tree.Node;
-import io.ballerina.compiler.syntax.tree.NodeVisitor;
-import io.ballerina.compiler.syntax.tree.NonTerminalNode;
-import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
-import io.ballerina.compiler.syntax.tree.RemoteMethodCallActionNode;
-import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
-import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.*;
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java:20]: Do not use the wildcard imports

File: misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java [hunk 2]
```diff
@@ -69,42 +68,33 @@
...
-        Position startPos = new Position(bLangNode.pos.lineRange().startLine().line() - 1,
-                bLangNode.pos.lineRange().startLine().offset() - 1);
-        Position endPosWithSemiColon = new Position(bLangNode.pos.lineRange().endLine().line() - 1,
-                bLangNode.pos.lineRange().endLine().offset());
-        Range newTextRange = new Range(startPos, endPosWithSemiColon);
-
-        BSymbol symbolAtCursor = refAtCursor.getSymbol();
+        Range newTextRange = diagnostic.getRange();
+
+        String symbolAtCursor = context.workspace().semanticModel(context.filePath()).get().symbol(context.filePath().getFileName().toString(), LinePosition.from(context.getCursorPosition().getLine(), context.getCursorPosition().getCharacter())).get().name();
```
Comment (nadeeshaan) [inline misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java:95]: This seems to be longer than 120 chars. Please format properly

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java [hunk 3]
```diff
@@ -114,6 +101,12 @@ public void visit(ClassDefinitionNode classDefinitionNode) {
         this.currentIdentifierPos = classDefinitionNode.className().lineRange().startLine();
     }
 
+    @Override
+    public void visit(VariableDeclarationNode classDefinitionNode) {
```
[the remark this answers is not in the document]
Comment (isuruRuhu) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java]: Change the parameter name to `variableDeclarationNode`

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java (hunk 1 above)
Comment (isuruRuhu) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/ScopedSymbolFinder.java]: Changed the code and doing the imports separately.

File: misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java (hunk 2 above)
Comment (isuruRuhu) [reply misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java]: Formated the code.

**Comment by isuruRuhu** [comment]
Sent a PR to the master 
https://github.com/ballerina-platform/ballerina-lang/pull/27083

**Pull request closed without merging**